src/main/java/org/jenkinsci/plugins/workflow/cps/CpsWhitelist.java

```diff
@@ -103,6 +104,9 @@ public boolean permitsStaticMethod(Method method, Object[] args) {
             String driverTo = "org.codehaus.groovy.runtime.";
             if (cn.startsWith(driverFrom)) {
                 try {
+                    if (Modifier.isPrivate(method.getModifiers())) {
+                        return true;
```

Comment:
So that is not right. The compiled code is carrying over the same `Env` as its caller. I wonder if we need to create a subclass of `CpsFunction` which passes a `ProxyEnv` with a `DefaultInvoker`.